### ISSUE TITLE
Open  "view only" mode failed

### ISSUE THREAD

**Issue by sniperking1234** [kiali/kiali#1215]
I edited kiali  CR and added `view_only_mode: true` config. But there was an error.

Kiali version:0.21
Kubernetes version:v1.11.1

**kiali CR:**
```
apiVersion: kiali.io/v1alpha1
kind: Kiali
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"kiali.io/v1alpha1","kind":"Kiali","metadata":{"annotations":{},"name":"kiali","namespace":"kiali-operator"},"spec":{"auth":{"strategy":"login"},"deployment":{"image_version":"v0.21.0","namespace":"istio-system","secret_name":"kiali"},"external_services":{"grafana":{"url":""},"jaeger":{"url":""}}}}
  creationTimestamp: 2019-06-17T05:19:09Z
  finalizers:
  - finalizer.kiali
  generation: 13
  name: kiali
  namespace: kiali-operator
  resourceVersion: "441309738"
  selfLink: /apis/kiali.io/v1alpha1/namespaces/kiali-operator/kialis/kiali
  uid: 6f6578a5-90bf-11e9-bf4a-6c92bf7360cc
spec:
  auth:
    strategy: login
  deployment:
    image_version: v0.21.0
    namespace: istio-system
    secret_name: kiali
    view_only_mode: true
  external_services:
    grafana:
      url: ""
    jaeger:
      url: ""
status:
  conditions:
  - lastTransitionTime: 2019-07-04T07:25:23Z
    message: Running reconciliation
    reason: Running
    status: "False"
    type: Running
  - ansibleResult:
      changed: 1
      completion: 2019-07-04T07:27:49.502137
      failures: 1
      ok: 33
      skipped: 16
    lastTransitionTime: 2019-07-04T07:27:49Z
    message: |
      Failed to patch object: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"ClusterRoleBinding.rbac.authorization.k8s.io \"kiali\" is invalid: roleRef: Invalid value: rbac.RoleRef{APIGroup:\"rbac.authorization.k8s.io\", Kind:\"ClusterRole\", Name:\"kiali-viewer\"}: cannot change roleRef","reason":"Invalid","details":{"name":"kiali","group":"rbac.authorization.k8s.io","kind":"ClusterRoleBinding","causes":[{"reason":"FieldValueInvalid","message":"Invalid value: rbac.RoleRef{APIGroup:\"rbac.authorization.k8s.io\", Kind:\"ClusterRole\", Name:\"kiali-viewer\"}: cannot change roleRef","field":"roleRef"}]},"code":422}
    reason: Failed
    status: "True"
    type: Failure
```
**operator ansible logs:**
> TASK [kiali-deploy : Create resource [clusterrolebinding] on [kubernetes]] *****
task path: /opt/ansible/roles/kiali-deploy/tasks/process-resource.yml:1
FAILED - RETRYING: Create resource [clusterrolebinding] on [kubernetes] (6 retries left).
FAILED - RETRYING: Create resource [clusterrolebinding] on [kubernetes] (5 retries left).
FAILED - RETRYING: Create resource [clusterrolebinding] on [kubernetes] (4 retries left).
FAILED - RETRYING: Create resource [clusterrolebinding] on [kubernetes] (3 retries left).
FAILED - RETRYING: Create resource [clusterrolebinding] on [kubernetes] (2 retries left).
FAILED - RETRYING: Create resource [clusterrolebinding] on [kubernetes] (1 retries left).
fatal: [localhost]: FAILED! => {"attempts": 6, "changed": false, "error": 422, "msg": "Failed to patch object: {\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"ClusterRoleBinding.rbac.authorization.k8s.io \\\"kiali\\\" is invalid: roleRef: Invalid value: rbac.RoleRef{APIGroup:\\\"rbac.authorization.k8s.io\\\", Kind:\\\"ClusterRole\\\", Name:\\\"kiali-viewer\\\"}: cannot change roleRef\",\"reason\":\"Invalid\",\"details\":{\"name\":\"kiali\",\"group\":\"rbac.authorization.k8s.io\",\"kind\":\"ClusterRoleBinding\",\"causes\":[{\"reason\":\"FieldValueInvalid\",\"message\":\"Invalid value: rbac.RoleRef{APIGroup:\\\"rbac.authorization.k8s.io\\\", Kind:\\\"ClusterRole\\\", Name:\\\"kiali-viewer\\\"}: cannot change roleRef\",\"field\":\"roleRef\"}]},\"code\":422}\n", "reason": "Unprocessable Entity", "status": 422}

**Comment by jmazzitelli** [comment]
Interesting... looks like you can set the roleRef when you CREATE a Role, but you can't PATCH a roleRef for an already existing role.

If that is true (and there is nothing else going on that I can't see), it looks like the operator is going to need to do some special tricks to delete the role if it already exists and then create an entirely new one when it sees the roleRef needs to change. Which is annoying.

**Comment by jmazzitelli** [comment]
https://github.com/kubernetes/kubernetes/issues/66353#issuecomment-408559096

So it is by design you cannot patch existing role bindings but instead must do a delete/create.

We'll need to change the operator to support changing this view_only_mode.

**Comment by jmazzitelli** [comment]
See #1221 for a potential fix. Has not been tested - would need someone to review/test.

**Comment by jmazzitelli** [comment]
this is fixed via #1221
